lms/djangoapps/mobile_api/users/tests.py

```diff
@@ -163,154 +94,134 @@ def _setup_course_skeleton(self):
 
-        url = self._course_status_url()
-        result = self.client.get(url)
-        json_data = json.loads(result.content)
+class TestCourseStatusGET(CourseStatusAPITestCase, MobileAuthUserTestMixin, MobileEnrolledCourseAccessTestMixin):
+    """
+    Tests for GET of /api/mobile/v0.5/users/<user_name>/course_status_info/
+    """
+    def test_success(self):
+        self.login_and_enroll()
+        (section, sub_section, unit, __) = self._setup_course_skeleton()
 
-        self.assertEqual(result.status_code, 200)
-        self.assertEqual(json_data["last_visited_module_id"], unicode(unit.location))
+        response = self.get_response()
+        self.assertEqual(response.data["last_visited_module_id"], unicode(unit.location))
         self.assertEqual(
-            json_data["last_visited_module_path"],
+            response.data["last_visited_module_path"],
             [unicode(module.location) for module in [unit, sub_section, section, self.course]]
         )
 
-    def test_course_update_no_args(self):
-        self.client.login(username=self.username, password=self.password)
 
-        url = self._course_status_url()
-        result = self.client.patch(url)  # pylint: disable=no-member
-        self.assertEqual(result.status_code, 200)
+class TestCourseStatusPATCH(CourseStatusAPITestCase, MobileAuthUserTestMixin, MobileEnrolledCourseAccessTestMixin):
+    """
+    Tests for PATCH of /api/mobile/v0.5/users/<user_name>/course_status_info/
+    """
+    def url_method(self, url, **kwargs):
+        # override implementation to use PATCH method.
+        return self.client.patch(url, data=kwargs.get('data', None))  # pylint: disable=no-member
 
-    def test_course_update(self):
+    def test_success(self):
+        self.login_and_enroll()
         (__, __, __, other_unit) = self._setup_course_skeleton()
-        self.client.login(username=self.username, password=self.password)
 
-        url = self._course_status_url()
-        result = self.client.patch(  # pylint: disable=no-member
-            url,
-            {"last_visited_module_id": unicode(other_unit.location)}
-        )
-        self.assertEqual(result.status_code, 200)
-        result = self.client.get(url)
-        json_data = json.loads(result.content)
-        self.assertEqual(result.status_code, 200)
-        self.assertEqual(json_data["last_visited_module_id"], unicode(other_unit.location))
-
-    def test_course_update_bad_module(self):
-        self.client.login(username=self.username, password=self.password)
-
-        url = self._course_status_url()
-        result = self.client.patch(  # pylint: disable=no-member
-            url,
-            {"last_visited_module_id": "abc"},
-        )
-        json_data = json.loads(result.content)
-        self.assertEqual(result.status_code, 400)
-        self.assertEqual(json_data, errors.ERROR_INVALID_MODULE_ID)
+        response = self.get_response(data={"last_visited_module_id": unicode(other_unit.location)})
```

Comment:
I found it kind of confusing that this thing was called ``get...`` but the HTTP method was overridden so it was actually a PATCH. Maybe there's a better name so it's more obvious whether it's it's get like a getter or get like a GET request.